src/main/java/com/artipie/npm/proxy/HttpNpmRemote.java

```diff
@@ -0,0 +1,176 @@
+        this.vertx = vertx;
+        this.client = WebClient.create(vertx, this.defaultWebClientOptions());
+    }
+
+    @Override
+    //@checkstyle ReturnCountCheck (40 lines)
+    public Maybe<NpmPackage> loadPackage(final String name) {
+        return this.client.getAbs(String.format("%s/%s", this.config.url(), name))
+            .timeout(this.config.requestTimeout())
+            .rxSend()
+            .flatMapMaybe(
+                response -> {
+                    //@checkstyle MagicNumberCheck (1 line)
+                    if (response.statusCode() == 200) {
+                        return Maybe.just(
+                            new NpmPackage(
+                                name,
+                                new CachedContent(response.bodyAsString(), name).value(),
+                                response.getHeader("Last-Modified")
+                            )
+                        );
+                    } else {
+                        Logger.debug(
+                            NpmProxy.class,
+                            "Could not load package: status code %d",
+                            response.statusCode()
+                        );
+                        return Maybe.empty();
+                    }
+                }
+            ).onErrorResumeNext(
+                throwable -> {
+                    Logger.error(
+                        NpmProxy.class,
+                        "Error occurred when process get package call: %s",
+                        throwable.getMessage()
+                    );
+                    return Maybe.empty();
+                }
+            );
+    }
+
+    @Override
+    //@checkstyle ReturnCountCheck (50 lines)
+    public Maybe<NpmAsset> loadAsset(final String path, final Path tmp) {
+        return this.vertx.fileSystem().rxOpen(
+            tmp.toAbsolutePath().toString(),
+            new OpenOptions().setSync(true).setTruncateExisting(true)
+        ).flatMapMaybe(
+            asyncfile ->
+                this.client.getAbs(
+                    String.format("%s/%s", this.config.url(), path)
+                ).as(
+                    BodyCodec.pipe(asyncfile)
+                ).rxSend().flatMapMaybe(
+                    response -> {
+                        // @checkstyle MagicNumberCheck (1 line)
+                        if (response.statusCode() == 200) {
+                            return Maybe.just(
+                                new NpmAsset(
+                                    path,
+                                    new RxFile(
+                                        tmp,
+                                        this.vertx.fileSystem()
+                                    ).flow(),
+                                    response.getHeader("Last-Modified"),
+                                    response.getHeader("Content-Type")
+                                )
+                            );
+                        } else {
```

Comment:
@g4s8 I don't think that HTTP 404 should be treated as exceptional state. We have tried to find something, we have failed - OK, we have the other options. Really, it does not matter at this moment what exactly occurred - connection reset, HTTP 500, HTTP 404 or something else. But if we will throw exceptional case, then we need to get it in the NpmProxy. And we need to handle in different ways NpmRemote errors and NpmStorage errors.
But maybe latter it will make sense. I have an issue to implement remote registry health check. In this case we will need to different actions depend on what exactly occurs.